internal/app/auditd/auditd.go

```diff
@@ -0,0 +1,51 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// Package auditd registers auditd service and logs audit events.
+package auditd
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/elastic/go-libaudit/v2"
+	"github.com/elastic/go-libaudit/v2/auparse"
+
+	"github.com/siderolabs/talos/internal/app/machined/pkg/runtime"
+)
+
+func Main(ctx context.Context, _ runtime.Runtime, logWriter io.Writer) error {
+	return Run(ctx, logWriter)
+}
+
+func Run(ctx context.Context, logWriter io.Writer) error {
+	var writer io.Writer
+
+	client, err := libaudit.NewMulticastAuditClient(writer)
+	if err != nil {
+		return fmt.Errorf("failed to create audit client: %w", err)
+	}
```

Comment:
NewMulticastAuditClient creates a readonly client, which cannot "claim" the audit subsystem, so messages will duplicate to printk, which is not really great. `SetPID` claims we are the only one audit operator, and `SetBacklogLimit` ensures log buffer is sufficient in case of peaks of activity

```suggestion
	client, err := libaudit.NewAuditClient(writer)
	if err != nil {
		return fmt.Errorf("auditd: failed to create client: %w", err)
	}

	if err := client.SetEnabled(true, libaudit.WaitForReply); err != nil {
		return fmt.Errorf("auditd: failed to enable: %w", err)
	}

	if err := client.SetPID(libaudit.NoWait); err != nil {
		return fmt.Errorf("auditd: failed to set PID: %w", err)
	}

	if err := client.SetBacklogLimit(4096, libaudit.NoWait); err != nil {
		return fmt.Errorf("auditd: failed to set backlog limit: %w", err)
	}
```

Reply:
yeh, let's do this, I just picked the easiest path

Reply:
Alternatively you can cherry-pick https://github.com/siderolabs/talos/pull/9617/commits/55e22ecf18d90f0b6dcb437c2d8627cbedd3ed87

Reply:
do we need ratelimiting?

Reply:
Unsure, maybe some high value like 4096 per sometimes as well. Do we store the log created by this service in a circular buffer, so we're not going to OOM due to it? Then I believe sth high could be okay, so that machined couldn't be DoSed just in case